### ISSUE TITLE
Training process fails to reproduce translation accuracy, while provided checkpoint works perfectly

### ISSUE THREAD

**Issue by DHW74** [CNJianLiu/MonoDiff9D#4]
<html>
<body>
<h3><strong><span class="selected"></span></strong></h3><p><span class="selected">Hello,</span></p><p><span class="selected">First of all, thank you for your excellent work on MonoDiff9D and for making the code publicly available! I am a graduate student researching 6D pose estimation and have been trying to reproduce the results from your paper.</span></p><p><span class="selected">I've encountered a puzzling situation that I hope you could help clarify.</span></p><h4><span class="selected">The Core Issue:</span></h4><p><strong><span class="selected">Your provided pretrained checkpoint (</span><code><span class="selected">epoch_210.pth</span></code><span class="selected">) works perfectly.</span></strong><span class="selected"> When I run the evaluation script (</span><code><span class="selected">test.py</span></code><span class="selected">) with your checkpoint, I can successfully reproduce the high-quality results reported in your paper for the REAL275 dataset.</span></p><p><span class="selected">However, when I train the model from scratch using the exact same codebase and configuration, my trained models consistently fail to match the </span><strong><span class="selected">translation accuracy</span></strong><span class="selected">. The rotation accuracy (</span><code><span class="selected">10°</span></code><span class="selected">) is nearly perfect, but all translation-related metrics are significantly lower. This happens whether I train for 210 epochs or 300+ epochs.</span></p><p><span class="selected">Here is a comparison between the results from my self-trained model and your paper's results (Table I, REAL275):</span></p>

<img width="508" height="247" alt="Image" src="https://github.com/user-attachments/assets/9a78e82d-87b5-4adb-b9e9-05deaae3913a" />

- 3D IoU at 50: 21.5
- 3D IoU at 75: 3.2
- 10 cm: 28.8
- 10 degree: 59.0
- 10 degree, 10cm: 15.8

<h4><span class="selected">Verified Steps:</span></h4><ul><li><p><strong><span class="selected">Environment:</span></strong><span class="selected"> My setup is built precisely from your </span><code><span class="selected">environment.yaml</span></code><span class="selected"> file. Since your checkpoint runs correctly, my environment and evaluation pipeline should be correct.</span></p></li><li><p><strong><span class="selected">Data Integrity:</span></strong><span class="selected"> I generated the depth maps using the prescribed DINOv2-NYU head. I performed a rigorous numerical comparison between my generated </span><code><span class="selected">.npy</span></code><span class="selected"> files and the samples you provided. The results confirmed they are highly consistent (correlation &gt; 0.9999, max absolute difference ~1e-2).</span></p></li></ul><h4><span class="selected">My Question:</span></h4><p><span class="selected">Could you please advise if there are any crucial details about the training procedure that might differ from the public code? For instance, specific learning rate schedules, weight initializations, or other hyperparameters that were used to train the successful </span><code><span class="selected">epoch_210.pth</span></code><span class="selected"> model?</span></p><p><span class="selected">Any insight you could provide would be incredibly helpful in understanding this gap.</span></p><p><span class="selected">Thank you again for your time and for this fantastic contribution to the community!</span></p>
</body>
</html>

**Comment by CNJianLiu** [comment]
Hi, thanks for your message. Did you get your depth map using our "../tools/depth_recovery.py" file? Actually I trained some times, and the results are close

**Comment by yumomo11** [comment]
Hello, senior. I am now a junior undergraduate student. I encountered some errors in the test and would like to seek your guidance. This project has been very helpful to me, and I sincerely hope to get in touch with you.

**Comment by DrLazywh** [comment]
result_pkl_list = glob.glob(os.path.join(self.data_dir, 'data', 'segmentation_results', 'CAMERA25', 'results_*.pkl'))
请问，能否提供一下CAMERA的测试集的segmentation_results，您提供的链接不包括这个CAMERA25的文件
